core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/version/VersionCreateRequest.java

```diff
@@ -114,24 +116,13 @@ public Boolean execute(RepositoryContext context) {
 		
 		if (!resourcesById.containsKey(resource)) {
 			context.log().warn("Resource cannot be found during versioning: " + resourcesById + ", uri: " + resource);
-			throw new NotFoundException("Resource", resource.getResourceId());
+			throw new BadRequestException("Resource", resource.getResourceId());
 		}
 		
 		// validate new path
 		context.service(BranchNameValidator.class).checkName(version);
 		
-		TerminologyResource resourceToVersion = resourcesById.get(resource);
-
-		if (TerminologyResourceCommitRequestBuilder.READ_ONLY_STATUSES.contains(resourceToVersion.getStatus())) {
-			throw new BadRequestException("Resource '%s' cannot be versioned in its current status '%s'", resourceToVersion.getTitle(), resourceToVersion.getStatus());
-		}
-		
-		// TODO resurrect or eliminate tooling dependencies
-		final List<TerminologyResource> resourcesToVersion = List.of(resourcesById.get(resource));
-//		final List<CodeSystem> resourcesToVersion = codeSystem.getDependenciesAndSelf()
-//				.stream()
-//				.map(resourcesById::get)
-//				.collect(Collectors.toList());
+		final Collection<TerminologyResource> resourcesToVersion = resourcesById.values();
 		
 		resourcesToVersion.stream()
 			.filter(cs -> cs.getUpgradeOf() != null)
```

Comment:
Is this check still required? Or shouldn't we do the same check using the new upgrade resource id?

Reply:
Not required anymore, removing it.